bee-network/src/service/controller.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    command::{Command, CommandSender},
+    event::{Event, EventReceiver},
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Error sending command.")]
+    CommandSendFailure,
+    #[error("Error receiving event.")]
+    EventRecvFailure,
+}
+
+/// Lets the user send [`Command`]s to the network layer.
+#[derive(Clone, Debug)]
+pub struct NetworkCommandSender(CommandSender);
+
+impl NetworkCommandSender {
+    pub(crate) fn new(inner: CommandSender) -> Self {
+        Self(inner)
+    }
+
+    /// Sends a command to the network.
+    /// NOTE: Although synchronous, this method never actually blocks.
+    pub fn send(&self, command: Command) -> Result<(), Error> {
+        self.0.send(command).map_err(|_| Error::CommandSendFailure)?;
```

Comment:
No need to throw error here, just return `self.0.send(command).map_err(|_| Error::CommandSendFailure)` directly

Reply:
done.  probably had a `trace` log in between there at some point. :smiley: